libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationView.kt

```diff
@@ -298,145 +424,169 @@ class NavigationView : ConstraintLayout {
                     routeArrowViewModel.routeProgressUpdated(routeProgress, style)
                 }
                 externalRouteProgressObservers.forEach {
-                    lifeCycleOwner.lifecycleScope.launch {
-                        it.onRouteProgressChanged(routeProgress)
+                    it.onRouteProgressChanged(routeProgress)
+                }
+                uiComponents.forEach { uiComponent ->
+                    when (uiComponent) {
+                        is ManeuverUIComponent.MapboxManeuverUIComponent -> {
+                            uiComponent.onRouteProgressChanged(routeProgress)
+                        }
+                        is TripProgressUIComponent.MapboxTripProgressUIComponent -> {
+                            uiComponent.onRouteProgressChanged(routeProgress)
+                        }
```

Comment:
```suggestion
                    when (uiComponent) {
                        is RouteProgressObserver -> {
                            uiComponent.onRouteProgressChanged(routeProgress)
                        }
```

Same below where appropriate.